packages/calling-component-bindings/src/handlers/createCommonHandlers.ts

```diff
@@ -0,0 +1,406 @@
+  onToggleCamera: (options?: VideoStreamOptions) => Promise<void>;
+  onSelectMicrophone: (device: AudioDeviceInfo) => Promise<void>;
+  onSelectSpeaker: (device: AudioDeviceInfo) => Promise<void>;
+  onSelectCamera: (device: VideoDeviceInfo, options?: VideoStreamOptions) => Promise<void>;
+  onToggleMicrophone: () => Promise<void>;
+  onStartScreenShare: () => Promise<void>;
+  onStopScreenShare: () => Promise<void>;
+  onToggleScreenShare: () => Promise<void>;
+  onHangUp: (forEveryone?: boolean) => Promise<void>;
+  /* @conditional-compile-remove(PSTN-calls) */
+  onToggleHold: () => Promise<void>;
+  /* @conditional-compile-remove(PSTN-calls) */
+  onAddParticipant(participant: CommunicationUserIdentifier): Promise<void>;
+  /* @conditional-compile-remove(PSTN-calls) */
+  onAddParticipant(participant: PhoneNumberIdentifier, options: AddPhoneNumberOptions): Promise<void>;
+  onCreateLocalStreamView: (options?: VideoStreamOptions) => Promise<void | CreateVideoStreamViewResult>;
+  onCreateRemoteStreamView: (
+    userId: string,
+    options?: VideoStreamOptions
+  ) => Promise<void | CreateVideoStreamViewResult>;
+  onDisposeRemoteStreamView: (userId: string) => Promise<void>;
+  onDisposeLocalStreamView: () => Promise<void>;
+  /* @conditional-compile-remove(dialpad) */ /* @conditional-compile-remove(PSTN-calls) */
+  onSendDtmfTone: (dtmfTone: DtmfTone) => Promise<void>;
+  onRemoveParticipant(userId: string): Promise<void>;
+  /* @conditional-compile-remove(PSTN-calls) */
+  onRemoveParticipant(participant: CommunicationIdentifier): Promise<void>;
+  /* @conditional-compile-remove(call-readiness) */
+  askDevicePermission: (constrain: PermissionConstraints) => Promise<void>;
+};
+
+/**
+ * Object containing all the handlers required for calling components.
+ *
+ * Calling related components from this package are able to pick out relevant handlers from this object.
+ * See {@link useHandlers} and {@link usePropsFor}.
+ *
+ * @public
+ */
+// CommonCallingHandlers is an interface compatible with TeamsCallingHandlers and CallingHandlers
+// which is also a common interface required by components and adapters
+// components and adapters only requires onStartCall is a function returning void
+// but in TeamsCallingHandlers and CallingHandlers, it returns different types for non-breaking change reason
+export type CommonCallingHandlers = SharedCallingHandlers & {
+  onStartCall: (
+    participants: (CommunicationUserIdentifier | PhoneNumberIdentifier | UnknownIdentifier)[],
+    options?: StartCallOptions
+  ) => void;
+};
+
+/**
+ * @private
+ */
+export const areStreamsEqual = (prevStream: LocalVideoStream, newStream: LocalVideoStream): boolean => {
+  return !!prevStream && !!newStream && prevStream.source.id === newStream.source.id;
+};
+
+/**
+ * Create the default implementation of {@link CallingHandlers}.
+ *
+ * Useful when implementing a custom component that utilizes the providers
+ * exported from this library.
+ *
+ * @public
```

Comment:
should this be public? Wouldn't Contoso either use createCallHandlers or createTeamsCallHandlers?

Reply:
Good catch, they don't need this, we don't actually export this in communication-react, lemme change the mark